### PR TITLE
chore(auth): update the example to offer the choice of using silent verification instead of Recaptcha

### PR DESCRIPTION
**Pull request by Lyokone**
## Description

More information here: https://firebase.google.com/docs/auth/ios/multi-factor#using_silent_notifications

## Related Issues

*Replace this paragraph with a list of issues related to this PR from the [issue database](https://github.com/flutter/flutter/issues). Indicate, which of these issues are resolved or fixed by this PR. Note that you'll have to prefix the issue numbers with flutter/flutter#.*

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [ ] I read and followed the [Flutter Style Guide].
- [ ] I signed the [CLA].
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [X] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/firebase/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Review by russellwheatley — APPROVED** [review]
Perhaps we ought to update the documentation as well for this?

**Comment by Lyokone** [comment]
> Perhaps we ought to update the documentation as well for this?

I'm waiting on Feedback to see if it's the proper behavior from the Firebase iOS SDK before updating the documentation

File: packages/firebase_auth/firebase_auth/example/lib/auth.dart [hunk 1]
```diff
@@ -15,6 +16,10 @@ typedef OAuthSignIn = void Function();
 
 final FirebaseAuth _auth = FirebaseAuth.instance;
 
+// If set to true, the app will request notification permissions to use
+// silent verification for SMS MFA instead of Recapcha.
```
Comment (Salakar) [inline packages/firebase_auth/firebase_auth/example/lib/auth.dart:20]:
```suggestion
// silent verification for SMS MFA instead of Recaptcha.
```

Comment (Lyokone) [reply packages/firebase_auth/firebase_auth/example/lib/auth.dart]: Fixed 👍

**Review by Salakar — APPROVED** [review]

**Pull request merged**